subcmd/edit.go

```diff
@@ -0,0 +1,218 @@
+package subcmd
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/vim-volt/volt/config"
+	"github.com/vim-volt/volt/lockjson"
+	"github.com/vim-volt/volt/logger"
+	"github.com/vim-volt/volt/pathutil"
+	"github.com/vim-volt/volt/subcmd/builder"
+)
+
+func init() {
+	cmdMap["edit"] = &editCmd{}
+}
+
+type editCmd struct {
+	helped bool
+	editor string
+}
+
+func (cmd *editCmd) ProhibitRootExecution(args []string) bool { return true }
+
+func (cmd *editCmd) FlagSet() *flag.FlagSet {
+	fs := flag.NewFlagSet(os.Args[0], flag.ContinueOnError)
+	fs.SetOutput(os.Stdout)
+	fs.Usage = func() {
+		fmt.Print(`
+Usage
+  volt edit [-help] [-e|--editor {editor}] {repository} [{repository2} ...]
+
+Quick example
+  $ volt edit tyru/caw.vim # will open the plugconf file for tyru/caw.vim for editing
+
+Description
+  Open the plugconf file(s) of one or more {repository} for editing.
+
+  If the -e option was given, use the given editor for editing those files (unless it cannot be found)
+
+  It also calls "volt build" afterwards if modifications were made to the plugconf file(s).` + "\n\n")
+		//fmt.Println("Options")
+		//fs.PrintDefaults()
+		fmt.Println()
+		cmd.helped = true
+	}
+	fs.StringVar(&cmd.editor, "editor", "", "Use the given editor for editing the plugconf files")
+	fs.StringVar(&cmd.editor, "e", "", "Use the given editor for editing the plugconf files")
+	return fs
+}
+
+func (cmd *editCmd) Run(args []string) *Error {
+	reposPathList, err := cmd.parseArgs(args)
+	if err == ErrShowedHelp {
+		return nil
+	}
+	if err != nil {
+		return &Error{Code: 10, Msg: "Failed to parse args: " + err.Error()}
+	}
+
+	hasChanges, err := cmd.doEdit(reposPathList)
+	if err != nil {
+		//FIXME: Which error code to use?
```

Comment:
An arbitrary code number is OK if it is unique in this `volt edit` command.

An error code is for debug purpose.
I agree it should be documented (for when error message is not visible...?).
It's not necessary in this PR, yes, sometime...